lib/sycamore/sycamore/utils/markdown.py

```diff
@@ -0,0 +1,152 @@
+"""
+Utilities for converting a list of Elements into Markdown-formatted text.
+
+TODO:
+- maybe insert horizontal rules at page breaks
+- handle numbered lists
+"""
+
+from typing import cast
+
+from sycamore.data import Element, TableElement
+
+
+SKIP_TYPES = {"page-header", "page-footer", "image"}
+
+
+def elements_to_markdown(elems: list[Element]) -> str:
+    """
+    This is the main function of interest.
+    Assumes elements are sorted as per bbox_sort.bbox_sort_document().
+    """
+    label_lists(elems)
+    s = ""
```

Comment:
eric would probably point out that building a string like this is n^2 - and it might actually make a difference here if I try to turn a textbook into markdown

Reply:
Yes, you are better off using StringIO, I would think.

Reply:
I forgot that Python doesn't have growable strings.  I can tweak this.  I have already run the 582-page SLO budget through without obvious issues, but have not tried the calculus text yet.